src/copaw/app/routers/upload.py

```
@@ -0,0 +1,176 @@
+        with open(dest, "wb") as f:
+            while chunk := await file.read(1024 * 1024):
+                size += len(chunk)
+                if size > MAX_FILE_SIZE:
+                    dest.unlink(missing_ok=True)
+                    raise HTTPException(
+                        status_code=413,
+                        detail=(
+                            f"File too large "
+                            f"(max {MAX_FILE_SIZE // (1024 * 1024)} MB)"
+                        ),
+                    )
```

Ruff (0.15.4)

[warning] Abstract `raise` to an inner function

(TRY301)